mcos/covariance_transformer.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+import numpy as np
+from sklearn.neighbors import KernelDensity
+from scipy.optimize import minimize
+import pandas as pd
+
+
+class AbstractCovarianceTransformer(ABC):
+    """
+    Abstract class for transforming a covariance matrix
+    """
+
+    def __init__(self, cov: np.array, n_observations: int):
+        self.cov = cov
+        self.n_observations = n_observations
```

Comment:
Should we remove the `__init__` method and move the parameters to the `transform` method?  Might as well keep it similar to the optimizer.

Reply:
Yeah that makes more sense than how things were, I was passing in `self.cov` and `self.n_observations` to `_de_noise_covariance_matrix` so they really shouldn't be instance variables since they're not being used as such. This is changed now